### ISSUE TITLE
Cannot start Plinth when apache2 is not installed

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/Plinth#358]
If I try to run Plinth when apache2 is not installed, I get this error:
```
File "/usr/local/lib/python3.4/dist-packages/Plinth-0.7.2-py3.4.egg/plinth/module_loader.py", line 129, in _initialize_module
    init()
  File "/usr/local/lib/python3.4/dist-packages/Plinth-0.7.2-py3.4.egg/plinth/modules/shaarli/__init__.py", line 43, in init
    is_external=True, enabled=is_enabled())
  File "/usr/local/lib/python3.4/dist-packages/Plinth-0.7.2-py3.4.egg/plinth/modules/shaarli/__init__.py", line 48, in is_enabled
    return action_utils.webserver_is_enabled('shaarli')
  File "/usr/local/lib/python3.4/dist-packages/Plinth-0.7.2-py3.4.egg/plinth/action_utils.py", line 92, in webserver_is_enabled
    stderr=subprocess.STDOUT)
  File "/usr/lib/python3.4/subprocess.py", line 607, in check_output
    with Popen(*popenargs, stdout=PIPE, **kwargs) as process:
  File "/usr/lib/python3.4/subprocess.py", line 859, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.4/subprocess.py", line 1463, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'a2query'
```
Since Apache2 is not a requirement for Plinth, Plinth should handle this exception and continue its startup.

**Comment by nikhilrayaprolu** [comment]
```
def webserver_is_enabled(name, kind='config'):
    """Return whether a config/module/site is enabled in Apache."""
    option_map = {'config': '-c', 'site': '-s', 'module': '-m'}
    try:
        # Don't print anything on the terminal
        subprocess.check_output(['a2query', option_map[kind], name],
                                stderr=subprocess.STDOUT)
        return True
    except subprocess.CalledProcessError:
        return False
```

found that the error is being called from this function in action_utils.py . So should we include apache2 in our dependencies or can this call be changed in a way possible to make it independent of apache

**Comment by jvalleroy** [comment]
@nikhilrayaprolu I would suggest to check that apache is installed before running a2query. If apache is not installed, report that the web config is not enabled (and perhaps provide a warning message or log).

Then in the future, we could extend this to also support nginx, if it is installed in place of apache.

It would be nice if there was a common way to check this for all popular webservers. But I don't know of any tool that provides this.

**Comment by nikhilrayaprolu** [comment]
basically in action_utils we have functions to check whether apache2 is running or not.but I see that we dont have a function to check whether it is installed or not.so I got this link regarding how to check if apache is installed or not. http://superuser.com/questions/387948/how-can-i-determine-if-apache-is-installed-on-a-system 
Shall we write a function to find out this so that we can use it where ever required in the application.
so is it okay to use "dpkg --get-selections | grep apache" as mentioned in the above stackoverflow discussion

**Comment by jvalleroy** [comment]
@nikhilrayaprolu I think it can be even simpler, just run "which a2query" and check if there is any output.

**Comment by jvalleroy** [comment]
@nikhilrayaprolu Here's another method using python standard library:
```
>>> from distutils import spawn
>>> spawn.find_executable('a2query')
'/usr/sbin/a2query'
```

**Comment by jvalleroy** [comment]
Fixed in #528.
